staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta_test.go

```diff
@@ -498,3 +498,65 @@ func TestValidateAnnotations(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateImmutableField(t *testing.T) {
+	type TestCaseStruct struct {
+		FieldString1 string
+		FieldString2 string
+		FieldInt1    int
+	}
+	type TestCaseStruct2 struct {
+		FieldString1 string
+		FieldString2 string
+	}
+	successTestCases := [][]interface{}{
+		{TestCaseStruct{"aaa", "bbb", 1}, TestCaseStruct{"aaa", "bbb", 1}},
+		{TestCaseStruct{"", "bbb", 1}, TestCaseStruct{"", "bbb", 1}},
+		{TestCaseStruct{}, TestCaseStruct{}},
+		{1, 1},
+		{nil, nil},
+		{map[string]interface{}{}, map[string]interface{}{}},
+		{map[string]interface{}{"aa": "bb", "cc": "dd"}, map[string]interface{}{"cc": "dd", "aa": "bb"}},
+		{map[string]interface{}{"aa": TestCaseStruct{"aaa", "bbb", 1}}, map[string]interface{}{"aa": TestCaseStruct{"aaa", "bbb", 1}}},
+	}
+
+	failTestCases := [][]interface{}{
+		{TestCaseStruct{}, TestCaseStruct2{}},
```

Comment:
Do we have that in practice, comparison across different types?

Reply:
Well, I am not sure, but I think this function should be able to handle this situation properly (anyway it won't jeopardize anything